classes/services/PKPContextService.php

```diff
@@ -547,10 +551,27 @@ public function add($context, $request)
         $genreDao = DAORegistry::getDAO('GenreDAO'); /** @var GenreDAO $genreDao */
         $genreDao->installDefaults($context->getId(), $context->getData('supportedLocales'));
 
-        Repo::userGroup()->installSettings($context->getId(), 'registry/userGroups.xml');
-
-        $managerUserGroup = Repo::userGroup()->getByRoleIds([Role::ROLE_ID_MANAGER], $context->getId(), true)->firstOrFail();
-        Repo::userGroup()->assignUserToGroup($currentUser->getId(), $managerUserGroup->getId());
+        $userGroupRepository = app(UserGroupRepository::class);
+        $userGroupRepository->installSettings($context->getId(), 'registry/userGroups.xml');
+
+
+        $managerUserGroup = UserGroup::where('contextId', $context->getId())
+            ->where('roleId', Role::ROLE_ID_MANAGER)
+            ->where('isDefault', true)
+            ->firstOrFail();
+        $assignmentExists = UserUserGroup::where('userId', $currentUser->getId())
```

Comment:
i think It's good to check if the assignment exists to prevent duplicate entries and ensure data integrity. Even though we're creating a new context, it's safer to include this check.